lib/libvfio-user.c

```diff
@@ -853,10 +896,29 @@ do_device_reset(vfu_ctx_t *vfu_ctx, vfu_reset_type_t reason)
     return 0;
 }
 
+static int
+handle_device_reset_quiesced(vfu_ctx_t *vfu_ctx, vfu_reset_type_t reason)
+{
+    return do_device_reset(vfu_ctx, reason);
+}
+
 int
 handle_device_reset(vfu_ctx_t *vfu_ctx, vfu_reset_type_t reason)
 {
-    return do_device_reset(vfu_ctx, reason);
+    int ret;
```

Comment:
we need async FLR support too.

Reply:
no - SPDK needs async reset, and this is one of the modes, so we need to do it

Reply:
I don't think I undertstand, we have two kinds of resets:

1. `handle_device_reset`, initiated by `VFIO_USER_DEVICE_RESET`, and
2. `vfu_reset_ctx`.

They can both be asynchronous. What am I missing?

Reply:
https://github.com/nutanix/libvfio-user/blob/master/lib/pci_caps.c#L286

Reply:
Note: FLR is initiated by writing to a PCI config space register (like migration state transition).

Reply:
I've added a check in the relevant unit test that the quiesce callback must be called and this fails now, so we won't miss it.